eden/scm/edenscm/git.py

```diff
@@ -142,7 +143,24 @@ def clone(ui, url, destpath=None, update=True, pullnames=None):
 
                 if not nodes:
                     pullnames.append(update)
-
+            else:
+                b = callgitnorepo(ui, ["ls-remote", "--symref", url, "HEAD"])
+                def parse_symref_head(b):
```

Comment:
I'm sure there is a better way to do this. Happy to make this cleaner, move the logic to a class, etc based on suggestions. 

At first I tried to use the [RefName](https://github.com/facebook/sapling/blob/5f017b9a2080de77130458d7168d23a886586432/eden/scm/edenscm/git.py#L273) class but it doesn't seem like there is any existing method that would yield the default branch name from the full ref.

Reply:
a(n unhelpful) review of your PR is at https://www.twitch.tv/videos/1697938402?t=03h52m12s

Reply:
I don't know whether to feel embarrassed or proud that someone live reviewed my PR on Twitch 😂 😳 😅

Reply:
(@vegerot BTW I am curious but have no context as to why someone asked it to be reviewed this stream? It wasn't me haha as I was not watching. Anyway, thanks! Or was it [strager](https://github.com/strager)? And yes I only meant that I thought there might be a better way to parse text 😆)

Reply:
umm, since the stream is http://twitch.tv/strager, I'd assume @strager was the reviewer